code/lib/cli/src/js-package-manager/parsePackageInfo.ts

```diff
@@ -0,0 +1,222 @@
+import sort from 'semver/functions/sort';
+
+const parsePackageData = (packageName = '') => {
+  const [first, second, third] = packageName.trim().split('@');
+  const version = (third || second).replace('npm:', '');
+  const name = third ? `@${second}` : first;
+
+  const value = { version, location: '' };
+  return { name, value };
+};
+
+type Yarn1ListItem = {
+  name: string;
+  children: Yarn1ListItem[];
+};
+
+type Yarn1ListData = {
+  type: 'list';
+  trees: Yarn1ListItem[];
+};
+
+export type Yarn1ListOutput = {
+  type: 'tree';
+  data: Yarn1ListData;
+};
+
+type PnpmDependency = {
+  from: string;
+  version: string;
+  resolved: string;
+  dependencies?: PnpmDependencies;
+};
+
+type PnpmDependencies = {
+  [key: string]: PnpmDependency;
+};
+
+type PnpmListItem = {
+  dependencies: PnpmDependencies;
+  peerDependencies: PnpmDependencies;
+  devDependencies: PnpmDependencies;
+};
+
+export type PnpmListOutput = PnpmListItem[];
+
+type NpmDependency = {
+  version: string;
+  resolved?: string;
+  overridden?: boolean;
+  dependencies?: NpmDependencies;
+};
+
+type NpmDependencies = {
+  [key: string]: NpmDependency;
+};
+
+export type NpmListOutput = {
+  dependencies: NpmDependencies;
+};
+
+type PackageMetadata = { version: string; location?: string; reasons?: string[] };
+export type InstallationMetadata = {
+  dependencies: Record<string, PackageMetadata[]>;
+  duplicatedDependencies: Record<string, string[]>;
+  infoCommand: string;
+};
+
+export const mapDependenciesNpm = (input: NpmListOutput): InstallationMetadata => {
```

Comment:
Would it make sense to put this in `NPMProxy`? Is it reused between files?

Reply:
Done!